src/routes/_surfaces/workspace/explore/leaderboards/LeaderboardDisplay.svelte

```diff
@@ -1,105 +1,109 @@
-  import { getDimensionsByMetricsId } from "$lib/redux-store/dimension-definition/dimension-definition-readables";
-  import { toggleSelectedLeaderboardValueAndUpdate } from "$lib/redux-store/explore/explore-apis";
-  import { getMetricsExplorerById } from "$lib/redux-store/explore/explore-readables";
-  import type {
-    LeaderboardValues,
-    MetricsExplorerEntity,
-  } from "$lib/redux-store/explore/explore-slice";
   import {
-    getMeasureFieldNameByIdAndIndex,
-    getMeasuresByMetricsId,
-  } from "$lib/redux-store/measure-definition/measure-definition-readables";
-  import { store } from "$lib/redux-store/store-root";
+    invalidateMetricViewData,
+    useGetMetricViewMeta,
+    useGetMetricViewTotals,
+  } from "$lib/svelte-query/queries/metric-view";
   import {
     getScaleForLeaderboard,
     NicelyFormattedTypes,
     ShortHandSymbols,
   } from "$lib/util/humanize-numbers";
+  import { useQueryClient } from "@sveltestack/svelte-query";
   import { onDestroy, onMount } from "svelte";
-  import type { Readable } from "svelte/store";
   import Leaderboard from "./Leaderboard.svelte";
 
   export let metricsDefId: string;
   export let whichReferenceValue: string;
 
-  let metricsExplorer: Readable<MetricsExplorerEntity>;
-  $: metricsExplorer = getMetricsExplorerById(metricsDefId);
+  const queryClient = useQueryClient();
+
+  let metricsExplorer: MetricsExplorerEntity;
+  $: metricsExplorer = $metricsExplorerStore.entities[metricsDefId];
 
-  let dimensions: Readable<DimensionDefinitionEntity[]>;
-  $: dimensions = getDimensionsByMetricsId(metricsDefId);
+  // query the `/meta` endpoint to get the metric's measures and dimensions
+  $: metaQuery = useGetMetricViewMeta(metricsDefId);
+  $: dimensions = $metaQuery.data.dimensions;
+  $: measures = $metaQuery.data.measures;
 
-  let measureField: Readable<string>;
-  $: if ($metricsExplorer?.leaderboardMeasureId)
-    measureField = getMeasureFieldNameByIdAndIndex(
-      $metricsExplorer.leaderboardMeasureId,
-      $metricsExplorer.measureIds.indexOf(
-        $metricsExplorer?.leaderboardMeasureId
-      )
+  $: activeMeasure =
+    measures &&
+    measures.find(
+      (measure) => measure.id === metricsExplorer?.leaderboardMeasureId
     );
 
-  $: measures = getMeasuresByMetricsId(metricsDefId);
-  $: leaderboardMeasureDefinition = $measures.find(
-    (measure) => measure.id === $metricsExplorer?.leaderboardMeasureId
-  );
-  // get the expression so we can determine if the measure is summable
-  $: expression = leaderboardMeasureDefinition?.expression;
   $: formatPreset =
-    leaderboardMeasureDefinition?.formatPreset ?? NicelyFormattedTypes.HUMANIZE;
+    activeMeasure?.formatPreset ?? NicelyFormattedTypes.HUMANIZE;
 
-  let bigNumberEntity: Readable<BigNumberEntity>;
-  $: bigNumberEntity = getBigNumberById(metricsDefId);
   let referenceValue: number;
 
-  $: if ($bigNumberEntity && $measureField) {
+  $: totalsQuery = useGetMetricViewTotals(
```

Comment:
why is `totalsQuery` reactive while `referenceValueQuery` is not? This is a confusing pattern.

Reply:
I think `referenceValueQuery` is supposed to be reactive too. @AdityaHegde touched that so I'll just let him confirm.

Reply:
Fixed!